### ISSUE TITLE
LCM: get Albany working with newer version of DataTransferKit (v3 or master branch) 

### ISSUE THREAD

**Issue by ikalash** [SNLComputation/Albany#543]
This would facilitate transitioning Albany with DTK to work with a Static graph - currently Albany works with DTK v. 2.0, not newer versions. I tried building Trilinos/Albany with a newer DTK, and there are compilation errors which seem to have nothing to do with our usage of DTK, rather there seems to be an issue building Trilinos with DTK.  I have opened a DTK issue to request some help with this: https://github.com/ORNL-CEES/DataTransferKit/issues/565 .

**Comment by ikalash** [comment]
I looked at DTK v3 and newer, as well as corresponded with some DTK developers about the prospects of switching Albany to a newer DTK (see https://github.com/ORNL-CEES/DataTransferKit/issues/565).  Unfortunately, it would not be straightforward.  The main reason is that all the adapters that were present in DTK v2 (in particular, the STK adapter), were removed in subsequent versions.  We rely heavily on the STK adapters in our current implementation in Albany - we grab all the necessary fields from STK and give them to DTK as STK objects.  To get the code to work with a newer DTK, we'd need to change all that code to convert the STK objects into other data structures that DTK expects now.  Since STK is far from simple, I imagine that this would be a pretty big headache (at least for a non-STK expert)...  

I've fixed Albany to work with DTK v2 by creating a fork of DTK v2 with some minor mods (see #544 ) - the fork can be found here: https://github.com/ikalash/DataTransferKit/tree/dtk-2.0-tpetra-static-graph .  Given that we have limited $$ for Schwarz development in Albany these days, probably we can defer upgrading DTK at this point...  but I am afraid at some point we may need to if Trilinos advances in ways that are not backward-compatible with DTK v2 and it's not possible to modify DTK v2 to work with those changes.  I guess we can cross that bridge when we get there...

@lxmota , feel free to comment on your thoughts, as a fellow user of DTK in Albany.
